main.go

```diff
@@ -549,3 +556,88 @@ func uploadFile(uploadURL string, archiveFilePath string) error {
 
 	return nil
 }
+
+func createDimensions(step toolresults.Step) map[string]string {
+	dimensions := map[string]string{}
+	for _, dimension := range step.DimensionValue {
+		dimensions[dimension.Key] = dimension.Value
+	}
+	return dimensions
+}
+
+func printStepsStatesToStartTime(stepsStatesToStartTime map[string]map[string]time.Time, stepsToNames map[string]string, currentTime time.Time, w io.Writer) {
+	var stepIDs []string
+	for stepID := range stepsToNames {
+		stepIDs = append(stepIDs, stepID)
+	}
+
+	sort.Strings(stepIDs)
+
+	for _, stepID := range stepIDs {
+		stepName := stepsToNames[stepID]
+		if _, err := fmt.Fprintln(w, stepName); err != nil {
+			fmt.Printf("Failed to print step status durations: %s", err)
+			return
+		}
+
+		statesToStartTime := stepsStatesToStartTime[stepID]
+
+		var states []string
+		for state := range statesToStartTime {
+			states = append(states, state)
+		}
+
+		sort.Slice(states, func(i, j int) bool {
+			stateI, stateJ := states[i], states[j]
+			startTimeI, startTimeJ := statesToStartTime[stateI], statesToStartTime[stateJ]
+
+			return startTimeI.Before(startTimeJ)
+		})
+
+		for i, state := range states {
+			startTime := statesToStartTime[state]
+
+			var endTime time.Time
+			if i == len(states)-1 {
+				endTime = currentTime
+			} else {
+				nextState := states[i+1]
+				endTime = statesToStartTime[nextState]
+			}
+
+			duration := endTime.Sub(startTime)
+
+			if _, err := fmt.Fprintf(w, "- time spent in %s state: ~%ds\n", state, duration/time.Second); err != nil {
```

Comment:
NOTE: You can use duration.Round(), see https://pkg.go.dev/time#Duration.Round/.

Reply:
updated in: https://github.com/bitrise-steplib/steps-virtual-device-testing-for-ios/pull/36/commits/a84e9481e79b06fc954cabad5f824f2695fbe3b4